### PR TITLE
Adding changes to the master branc

### PR DESCRIPTION
**Pull request by alexasf**
Added assembly option for different board layout options hardware_pin.h
Added a project folder for IAR
Debugged calibration mode
Board type selection in config_8s003.h #define BOARD_HW_V3

### CODE REVIEW

File: src/gpio_8s003.c [hunk 1]
```diff
@@ -1,170 +1,190 @@
...
-    if (digit != 0) { SETBIT(PD_ODR, PIN4); } else { CLRBIT(PD_ODR, PIN4); }
-    if (digit != 1) { SETBIT(PD_ODR, PIN6); } else { CLRBIT(PD_ODR, PIN6); }
-    if (digit != 2) { SETBIT(PD_ODR, PIN5); } else { CLRBIT(PD_ODR, PIN5); }
-
-    if (digit != 3) { SETBIT(PB_ODR, PIN5); } else { CLRBIT(PB_ODR, PIN5); }
-    if (digit != 4) { SETBIT(PB_ODR, PIN4); } else { CLRBIT(PB_ODR, PIN4); }
-    if (digit != 5) { SETBIT(PA_ODR, PIN1); } else { CLRBIT(PA_ODR, PIN1); }
+#include "config.h"
+#include "gpio.h"
+#include "hardware_pin.h"
```
Comment (cocus) [inline src/gpio_8s003.c]: I think this file should be included from config_8s003.h, so we don't have scattered config files per each platform

**Comment by cocus** [comment]
I don't have my original hardware to test this; have you tried it with the old hardware to check if it still works? Otherwise I'll merge it and fix it later whenever I have access to my meter.

File: src/gpio_8s003.c (hunk 1 above)
Comment (alexasf) [reply src/gpio_8s003.c]: Ok, so let's move to config_8s003.h

**Comment by alexasf** [comment]
Yes I was debugging this on a real board.
Just didn't check the last rounding addition. But it should work as I used the standard bubble library.
For your source project, the BOARD_HW_V2 configuration

**Comment by cocus** [comment]
I'll go ahead and merge your changes, if I find a bug on my hardware revision, I'll commit fixes if needed. Thank you!

**Comment by cocus** [comment]
Yes. There were some issues because the usage of roundf and some other functions that are not available on sdcc, but might be availble on IAR C. Please have a look at https://github.com/cocus/vc288-alt-fw/runs/2261838540?check_suite_focus=true.

**Pull request merged**